### ISSUE TITLE
`outdated` is wrong

### ISSUE THREAD

**Issue by ryanbooker-anz** [mas-cli/mas#499]
## Your Environment

- `mas version`: 1.8.6
- macOS version (`system_profiler SPSoftwareDataType -detailLevel mini`): 13.4.1

### mas Install Method

- [x] `brew install mas` (homebrew-core)
- [ ] [mas-cli/tap](https://github.com/mas-cli/homebrew-tap)
- [ ] `.pkg` installer from [releases](https://github.com/mas-cli/mas/releases)
- [ ] Built from source
  - Fork/branch: ? (e.g. mas-cli/main)
  - Xcode version: 10.?

## Describe the Bug

`mas` incorrectly reports outdated apps.

## To Reproduce

1. mas install TestFlight
1. mas outdated
1. Note it shows a new version that doesn't exit

## Expected Behavior

> A clear and concise description of what you expected to happen.

It doesn't show an incorrect version as available

## Actual Behavior

> A clear and concise description of what actually happened.

It shows what appears to be a phantom version.

e.g. None of these versions exist in the store.
```
1613020051 Snippet (0.5.3 -> 1.0)
1365531024 1Blocker (4.1.14 -> 5.3.7)
1591303229 Vinegar (2.4.1 -> 2.4.2)
899247664 TestFlight (3.3.0 -> 3.3.2)
```

**Comment by caius** [comment]
I'm also seeing this, and can reproduce it when installing mas-cli via the tap as well.

```
$ mas version  
1.8.6

$ mas outdated 
6444602274 Ivory (1.0.1 -> 1.4.1)

$ system_profiler SPSoftwareDataType | grep -e Version -e Mode -e Enabled
      System Version: macOS 13.4.1 (22F82)
      Kernel Version: Darwin 22.5.0
      Boot Mode: Normal
      Secure Virtual Memory: Enabled
      System Integrity Protection: Enabled
```

**Comment by caius** [comment]
I wonder if this is a Universal binary issue actually, Ivory works on both macOS and iOS and looking on the App Store page it lists iOS 15.5 as the minimum version yet on macOS it's 13.0 or later. On macOS App Store it shows the latest release as 1.0.1 , but on the iOS App Store it lists 1.4.1 as the latest release.

**Comment by caius** [comment]
Looks like this is a duplicate of https://github.com/mas-cli/mas/issues/491 and there's a potential fix up at #496

**Comment by ryanbooker-anz** [comment]
@caius The macOS and iOS versions of Ivory are different apps…

**Comment by RintaroYamada** [comment]
Hello,

I've encountered a similar issue as described here.

**Environment**: 
- mas-cli version: 1.8.6
- macOS version: 13.5.1

**Issue**:
When I run `mas outdated`, it shows an update for `辞書 by 物書堂` from version 1.2.44 to 1.2.45. However, upon checking the App Store, the latest version available is 1.2.44.

**Reproduction**:
1. Run `mas outdated`.
2. Observe the reported version discrepancy for `辞書 by 物書堂`.

Is there any workaround or solution for this?

Thank you.

**Comment by rgoldberg** [comment]
Closing as a duplicate of #491. PR #496 fixed the issue. It will be released in the upcoming version 1.8.7, which is currently in beta.

If you think this issue is different than #491, please let us know.
